Source/ThirdParty/WPE-platform/src/westeros/WesterosViewbackendInput.cpp

```diff
@@ -138,24 +139,43 @@ void WesterosViewbackendInput::keyboardHandleRepeatInfo( void *userData, int32_t
 
 void WesterosViewbackendInput::handleKeyEvent(void* userData, uint32_t key, uint32_t state, uint32_t time)
 {
-    auto& backend_input = *static_cast<WesterosViewbackendInput*>(userData);
-    auto& handlerData = backend_input.m_handlerData;
+    struct KeyEventData
+    {
+        void* userData;
+        uint32_t key;
+        uint32_t state;
+        uint32_t time;
+    };
 
-    auto& xkb = handlerData.xkb;
-    uint32_t keysym = xkb_state_key_get_one_sym(xkb.state, key);
-    uint32_t unicode = xkb_state_key_get_utf32(xkb.state, key);
+    KeyEventData *eventData = new KeyEventData { userData, key, state, time };
 
-    if (xkb.composeState
-        && state == WL_KEYBOARD_KEY_STATE_PRESSED
-        && xkb_compose_state_feed(xkb.composeState, keysym) == XKB_COMPOSE_FEED_ACCEPTED
-        && xkb_compose_state_get_status(xkb.composeState) == XKB_COMPOSE_COMPOSED)
+    g_idle_add_full(G_PRIORITY_DEFAULT, [](gpointer data) -> gboolean
     {
-        keysym = xkb_compose_state_get_one_sym(xkb.composeState);
-        unicode = xkb_keysym_to_utf32(keysym);
-    }
+        KeyEventData *e = (KeyEventData*)data;
+
+        auto& backend_input = *static_cast<WesterosViewbackendInput*>(e->userData);
+        auto& handlerData = backend_input.m_handlerData;
```

Comment:
Yes, that's true, this can become a problem when the view lifetime isn't tied to the application.